document-store/src/main/java/org/hypertrace/core/documentstore/model/config/ConnectionConfig.java

```diff
@@ -38,11 +40,11 @@ public static ConnectionConfigBuilder builder() {
   @FieldDefaults(level = AccessLevel.PRIVATE)
   public static class ConnectionConfigBuilder {
     DatabaseType type;
-    String host = DEFAULT_HOST;
-    Integer port;
+    List<Endpoint> endpoints = new ArrayList<>();
```

Comment:
Did we update any service that uses it directly (the new API)? if, yes, this change will cause compiler error on that service on upgrade of library, right?

Reply:
> Did we update any service that uses it directly (the new API)? 

No.

> if, yes, this change will cause compiler error on that service on upgrade of library, right?

Even if so, it would not since the public methods haven't changed. Only the internal representation and the way it is consumed in the library has changed.